flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -254,7 +255,18 @@ public int hashCode() {
 								states.push(stateTransition.getTargetState());
 								break;
 							case IGNORE:
-								resultingComputationStates.add(computationState);
+								final DeweyNumber version;
+								if (branched) {
```

Comment:
Here we assume that the `IGNORE` is going to be processed after the `TAKE`. This is true because of the way the `NFACompiler` creates the pattern graph, but different implementations may change in the future so we should somehow impose this ordering. It could be by sorting the `stateTransitions`  list in the `State` class, or by sorting them just before processing them in the `NFA::computeNextStates()`.